### ISSUE TITLE
How to end process upon "bye"?

### ISSUE THREAD

**Issue by sun-ruiheng** [nus-cs2103-AY2324S2/forum#366]
How can I terminate my entire chatbot upon encountering the "bye" command in my input? It used to be done by simply breaking out of the while loop. But after implementing the GUI with JavaFX, I just can't seem to identify any methods provided that effectively kills the whole process.

I can only think of System.exit(0).

**Comment by gavingoh99** [comment]
hello ruiheng, i found this while doing my usual rounds through stackoverflow (on the JavaFX section) which I think would be quite useful for you!

https://stackoverflow.com/questions/46053974/using-platform-exit-and-system-exitint-together

the gist of it is that we can use Platform (provided by javafx) and call its exit method...

hope this was useful to you!

**Comment by sun-ruiheng** [comment]
Wow! Thanks @gavingoh99, I think that's a less drastic way to terminate compared to a System.exit(0), although they behave similarly to the naked eye.

I'd still be happy to hear how others choose to terminate their programs!

**Comment by baskargopinath** [comment]
@sun-ruiheng what does System.exit(0) do?

**Comment by itstrueitstrueitsrealitsreal** [comment]
For me, I just included a line in the method that reads in user input to check if user input equals to "bye" after I obtain my output, and triggers a delay to allow for printing of output, then triggers `System.exit(0)` to exit the program as follows:

```java
        if (input.trim().equals("bye") || input.trim().equals("bb")) {
            PauseTransition pause = new PauseTransition(Duration.seconds(1)); // Adjust the delay as needed
            pause.setOnFinished(event -> System.exit(0));
            pause.play();
        }
```

Hope this helps!

**Comment by itstrueitstrueitsrealitsreal** [comment]
> @sun-ruiheng what does System.exit(0) do?

[Googling](https://www.baeldung.com/java-system-exit) it brings up that it is used to exit the Java program with an exit code of 0 (no errors).

**Comment by ThatGirlJam** [comment]
Hi ruiheng, 

For myself I created an exit() method in my equivalent of the Duke class, that is called upon the keyword "bye". I forced the GUI to not close (when user tries to click "X" to close) unless the user's latest input is bye. I prompt the user to say bye using an Alert. 

It's definitely a roundabout way, and assumes the user manually terminated the program, but perhaps this could be an idea to consider. All the best!
